pandas/core/apply.py

```diff
@@ -179,15 +180,238 @@ def apply(
         """
 
 
+class NumbaExecutionEngine(BaseExecutionEngine):
+    """
+    Numba-based execution engine for pandas apply and map operations.
+    """
+
+    @staticmethod
+    def map(
+        data: np.ndarray | Series | DataFrame,
+        func,
+        args: tuple,
+        kwargs: dict,
+        decorator: Callable | None,
+        skip_na: bool,
+    ):
+        """
+        Elementwise map for the Numba engine. Currently not supported.
+        """
+        raise NotImplementedError(
+            "The Numba engine is not implemented for the map method yet."
+        )
+
+    @staticmethod
+    def apply(
+        data: np.ndarray | Series | DataFrame,
+        func,
+        args: tuple,
+        kwargs: dict,
+        decorator: Callable,
+        axis: int | str,
+    ):
+        """
+        Apply `func` along the given axis using Numba.
+        """
+
+        NumbaExecutionEngine.check_numba_support(func)
+
+        # normalize axis values
+        if axis in (0, "index"):
+            axis = 0
+        else:
+            axis = 1
+
+        # check for data typing
+        if not isinstance(data, np.ndarray):
+            if data.empty:
+                return data.copy()  # mimic apply_empty_result()
+
+            engine_kwargs = extract_numba_options(decorator)
+
+            NumbaExecutionEngine.validate_values_for_numba_raw_false(
+                data, get_jit_arguments(engine_kwargs)
```

Comment:
It'd be better if the function receives the arguments directly, not packed in a dictionary. Then you can **get_jit_arguments(...